### PR TITLE
Apply Doctrine CS 4.0, version composer.lock

### PR DESCRIPTION
**Pull request by Majkl578**
* Exclude abstract/exception naming sniff
  * Abstracts either need to be reworked separately or ignored.
  * Exceptions are being handled in #6743.
* Exclude EntityManagerInterface from interface naming sniff
  * Do we want to rename this in 3.0?
* Use inline comments for single `@var` annotations
* Use short type casts
* Remove empty comments
* Apply other fixes (and fix previous breakages in master)

-----

Controversially, after the discussions in/around #30, we should start versioning the composer.lock file and use it to run CS/SA checks on CI. I am still not convinced it's a good idea to have it in the root though, but Composer doesn't have a switch to specify alternative lock file path...

-----

CS stage green, PHPStan stage fixed in #7113.

### CODE REVIEW

**Review by carusogabriel — APPROVED** [review]

File: lib/Doctrine/ORM/Mapping/FieldMetadata.php [hunk 1]
```diff
@@ -18,21 +17,10 @@ class FieldMetadata extends LocalColumnMetadata implements Property
...
-     */
-    /*public function __construct(string $name, string $columnName, Type $type)
-    {
-        parent::__construct($columnName, $type);
-
-        $this->name = $name;
-    }*/
-
-    public function __construct(string $name)
+    public function __construct(string $name/*, string $columnName, Type $type*/)
```
Comment (Ocramius) [inline lib/Doctrine/ORM/Mapping/FieldMetadata.php]: Weird one - is there an issue or WIP for this, @guilhermeblanco?

File: phpcs.xml.dist [hunk 2]
```diff
@@ -20,6 +20,8 @@
         <exclude name="SlevomatCodingStandard.TypeHints.TypeHintDeclaration.MissingReturnTypeHint"/>
         <exclude name="SlevomatCodingStandard.Exceptions.ReferenceThrowableOnly.ReferencedGeneralException"/>
         <exclude name="SlevomatCodingStandard.ControlStructures.EarlyExit"/>
+        <exclude name="SlevomatCodingStandard.Classes.SuperfluousAbstractClassNaming"/>
+        <exclude name="SlevomatCodingStandard.Classes.SuperfluousExceptionNaming"/>
```
Comment (Ocramius) [inline phpcs.xml.dist]: @greg0ire relevant for you

**Review by Ocramius — APPROVED** [review]
LGTM :+1:

File: phpcs.xml.dist (hunk 2 above)
Comment (greg0ire) [reply phpcs.xml.dist]: How will that work :thinking: ? Will that warn on new classes only?

Comment (Ocramius) [reply phpcs.xml.dist]: Will run on everything

File: lib/Doctrine/ORM/Mapping/FieldMetadata.php (hunk 1 above)
Comment (guilhermeblanco) [reply lib/Doctrine/ORM/Mapping/FieldMetadata.php]: WIP

**Pull request merged**